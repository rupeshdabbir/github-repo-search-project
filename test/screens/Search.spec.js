import React from 'react';
import { shallow } from 'enzyme';
import SearchScreen from 'screens/Search';
import DocumentTitle from 'react-document-title';
import SearchResultsContainer from 'containers/SearchResults';

describe('Screen: SearchScreen', () => {
  it('should create a document title', () => {
    const props = {
      location: {
        search: '?per_page=42&page=1&q=simon',
      },
    };
    const wrapper = shallow(<SearchScreen {...props} />);
    expect(wrapper.find(DocumentTitle).props().title).toEqual('simon - Page 1 - Github Repo search');
  });

  it('should pass match props and searchTerm to SearchResultsContainer', () => {
    const props = {
      location: {
        search: '?per_page=42&page=1&q=simon',
      },
    };
    const wrapper = shallow(<SearchScreen {...props} />);
    expect(wrapper.find(SearchResultsContainer).props()).toMatchSnapshot();
  });
});
