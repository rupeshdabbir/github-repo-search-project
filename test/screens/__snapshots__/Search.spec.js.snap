// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen: SearchScreen should pass match props and searchTerm to SearchResultsContainer 1`] = `
Object {
  "location": Object {
    "search": "?per_page=42&page=1&q=simon",
  },
  "searchTerm": "simon",
}
`;
